js/src/config/ProtectedRoute.jsx

```diff
@@ -1,39 +1,46 @@
 import { useState, useEffect } from 'react';
 import { Navigate } from 'react-router-dom';
 import { fetchAuthSession } from 'aws-amplify/auth';
+import { useAuthStore } from '../store/useStore';
+import { defineUserAbility } from '../ability/defineAbility';
 import { LoadingSpinner } from '../components';
 
-const ProtectedRoute = ({ children }) => {
-  const [isAuthenticated, setIsAuthenticated] = useState(false);
+const ProtectedRoute = ({ children, role }) => {
+  const { user, setUser, ability, setAbility } = useAuthStore();
   const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     const checkAuth = async () => {
       try {
         const session = await fetchAuthSession();
-
         if (session?.tokens?.idToken) {
-          setIsAuthenticated(true);
+          const userRole = session.tokens.idToken.payload['custom:role'].toLowerCase(); // Convert to lowercase
+
+          if (!user) {
+            setUser({ role: userRole });
+          }
+
+          const newAbility = defineUserAbility(userRole);
+          setAbility(newAbility);
         } else {
-          console.warn('No valid session tokens found.');
-          setIsAuthenticated(false);
+          console.warn('⚠️ No valid session tokens found.');
         }
       } catch (err) {
-        console.error('Error while checking authentication:', err);
-        setIsAuthenticated(false);
+        console.error('❌ Error while checking authentication:', err);
       } finally {
         setLoading(false);
       }
     };
 
     checkAuth();
-  }, []);
+  }, [user, setUser, setAbility]);
+
+  if (loading) return <LoadingSpinner />;
 
-  if (loading) {
-    return <LoadingSpinner />;
-  }
+  if (!user) return <Navigate to="/login" replace />;
```

Comment:
I dont think it should be replace here. Check once
replace will lose the history

Reply:
removed replace.